tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,975 @@
+          m.chai.expect(latestVersion).to.be.undefined;
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given a single version', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([ '0.5.0' ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should resolve the version', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('0.5.0');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given multiple versions', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '2.1.0',
+          '1.0.0',
+          '0.5.0',
+          '0.4.0'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should resolve the latest version', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('2.1.0');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given a greater production version in a snapshot bucket', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '1.0.0+abb6139',
+          '2.0.0'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should ignore production versions', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.SNAPSHOT).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('1.0.0+abb6139');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given a greater snapshot version in a production bucket', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '1.0.0',
+          '2.0.0+abb6139'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should ignore snapshot versions', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('1.0.0');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given v1, v2, and v3 remote versions', function() {
```

Comment:
I guess this could be clarified to 'given production v1, v2, and v3 remote versions' ?